src/main/java/interface_adapter/student_home/StudentHomeState.java

```diff
@@ -14,10 +24,50 @@ public void setUsername(String username) {
         this.username = username;
     }
 
+    public String getEmail() {
+        return this.email;
+    }
+
+    public void setEmail(String email) {
+        this.email = email;
+    }
+
+    public String getQuery() {
+        return this.query;
+    }
+
+    public void setQuery(String query) {
+        this.query = query;
+    }
+
+    public String getStudentHomeError() {
+        return this.studentHomeError;
+    }
+
+    public void setStudentHomeError(String studentHomeError) {
+        this.studentHomeError = studentHomeError;
+    }
+
+    public ArrayList<Club> getFollowedClubs() {
+        return this.followedClubs;
+    }
+
+    public void setFollowedClubs(ArrayList<Club> followedClubs) {
+        this.followedClubs = followedClubs;
+    }
+
     @Override
     public String toString() {
         return "StudentHomeState{"
                 + "username='" + username + '\''
                 + '}';
     }
```

Comment:
Having the student's email inside here would help since we are using emails as identifiers. If this is not ever going to be used for testing, we can also remove the toString function.